drivers/network/tcpip/ip/transport/tcp/tcp.c

```diff
@@ -379,10 +381,10 @@ NTSTATUS TCPConnect
         }
 
         /* Allocate the port in the port bitmap */
-        Connection->AddressFile->Port = TCPAllocatePort(LocalAddress.Address[0].Address[0].sin_port);
-
+        AllocatedPort = TCPAllocatePort(LocalAddress.Address[0].Address[0].sin_port);
         /* This should never fail */
```

Comment:
/* This should never fail unless all ports are in use */

Reply:
```suggestion
        /* This should never fail unless all ports are in use */
```